### PR TITLE
Added new module for an embedded Kafka Connect server

### PR DESCRIPTION
**Pull request by mmolimar**
Added support for an embedded Kafka Connect server in the library in a new module.

Also, there are some tests included.

### CODE REVIEW

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedConnectConfig.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+  * Kafka Connect config generator.
+  *
+  * @tparam C an [[EmbeddedKafkaConfig]]
+  */
+private[embeddedkafka] trait EmbeddedConnectConfig[C <: EmbeddedKafkaConfig] {
+  protected[embeddedkafka] def baseConnectConfig(
+      connectPort: Int,
+      offsets: File
```
Comment (francescopellegrini) [inline kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedConnectConfig.scala:18]: I'd use `java.nio.file.Path` instead, as it's already used by the library sometimes; also it works nicely with `Files.createTempFile` ;)

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    * @param connectPort the Kafka Connect port for the REST API to listen on.
+    * @param offsets     the file to store offset data in.
+    * @param extraConfig additional Kafka Connect configuration (overwrite existing keys in
+    *                    default config)
+    * @param block       the code block that will executed while the Kafka Connect server
+    *                    is active. Once the block has been executed the server will be stopped.
+    */
+  def startConnect[T](
+      connectPort: Int,
+      offsets: File,
```
Comment (francescopellegrini) [inline kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala:56]: I'd use `java.nio.file.Path` instead, as it's already used by the library sometimes; also it works nicely with `Files.createTempFile` ;)

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+import scala.jdk.CollectionConverters._
+
+/**
+  * Helper trait for running Kafka Connect server.
+  * Use `.startConnect` to start the server.
+  */
+trait EmbeddedKafkaConnect
+    extends EmbeddedKafkaConnectSupport[EmbeddedKafkaConfig]
```
Comment (francescopellegrini) [inline kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala:30]: Shouldn't this type param be `EmbeddedConnectConfig` instead?

**Review by francescopellegrini — CHANGES_REQUESTED** [review]

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedConnectConfig.scala (hunk 1 above)
Comment (mmolimar) [reply kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedConnectConfig.scala]: Done!

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala (hunk 2 above)
Comment (mmolimar) [reply kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala]: Done!

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala (hunk 3 above)
Comment (mmolimar) [reply kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala:30]: I'd say it's fine. The same way as it is in the EmbeddedKafkaStreams trait

Comment (francescopellegrini) [reply kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala:30]: Yes, but Streams requires no additional config property, while Connect does. Schema Registry, for one, references its own config (see [here](https://github.com/embeddedkafka/embedded-kafka-schema-registry/blob/master/embedded-kafka-schema-registry/src/main/scala/net.manub.embeddedkafka/schemaregistry/EmbeddedKafka.scala#L13)).

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedConnectConfig.scala [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  )(
+      implicit kafkaConfig: C
+  ): Map[String, String] =
+    Map(
+      WorkerConfig.LISTENERS_CONFIG                        -> s"http://localhost:$connectPort",
+      WorkerConfig.BOOTSTRAP_SERVERS_CONFIG                -> s"localhost:${kafkaConfig.kafkaPort}",
+      WorkerConfig.KEY_CONVERTER_CLASS_CONFIG              -> "org.apache.kafka.connect.json.JsonConverter",
+      WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG            -> "org.apache.kafka.connect.json.JsonConverter",
+      WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG        -> "10000",
+      StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG -> offsets.toFile.getAbsolutePath
```
Comment (francescopellegrini) [inline kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedConnectConfig.scala:28]: `.toFile` conversion could be skipped by calling `.toAbsolutePath.toString`

**Review by francescopellegrini — CHANGES_REQUESTED** [review]

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala (hunk 3 above)
Comment (mmolimar) [reply kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala:30]: You mean adding the case class extending from ``EmbeddedKafkaConfig`` and extending its corresponding trait, right?

Comment (francescopellegrini) [reply kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedKafkaConnect.scala:30]: I'm sorry, I've just noticed that `EmbeddedConnectConfig` does not extend `EmbeddedKafkaConfig` (as Schema Registry config does), but it's a wrapper instead.

File: kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedConnectConfig.scala (hunk 4 above)
Comment (mmolimar) [reply kafka-connect/src/main/scala/net/manub/embeddedkafka/connect/EmbeddedConnectConfig.scala]: Done!

**Review by NeQuissimus — APPROVED** [review]

**Review by francescopellegrini — APPROVED** [review]

**Pull request merged**